src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -2578,670 +2620,1067 @@ void gfx_set_framebuffer(int fb, float noise_scale);
-            s0 = C1(16, 16);
-            t0 = C1(0, 16);
-            ++cmd;
-            x1 = C0(16, 16);
-            y1 = C0(0, 16);
-            s1 = C1(16, 16);
-            t1 = C1(0, 16);
-            gfx_dp_image_rectangle(tile, iw, ih, x0, y0, s0, t0, x1, y1, s1, t1);
-            break;
+    }
+
+    gfx_dp_set_texture_image(C0(21, 3), C0(19, 2), C0(0, 10), imgData, texFlags, rawTexMetdata, (void*)i);
+
+    return false;
+}
+
+// TODO handle special OTR opcodes later...
+bool gfx_set_timg_otr_hash_handler_custom(Gfx** cmd0) {
+    uintptr_t addr = (*cmd0)->words.w1;
+    (*cmd0)++;
+    uint64_t hash = ((uint64_t)(*cmd0)->words.w0 << 32) + (uint64_t)(*cmd0)->words.w1;
+
+    const char* fileName = ResourceGetNameByCrc(hash);
+    uint32_t texFlags = 0;
+    RawTexMetadata rawTexMetadata = {};
+
+    std::shared_ptr<LUS::Texture> texture = std::static_pointer_cast<LUS::Texture>(
+        LUS::Context::GetInstance()->GetResourceManager()->LoadResourceProcess(ResourceGetNameByCrc(hash)));
+    if (texture != nullptr) {
+        texFlags = texture->Flags;
+        rawTexMetadata.width = texture->Width;
+        rawTexMetadata.height = texture->Height;
+        rawTexMetadata.h_byte_scale = texture->HByteScale;
+        rawTexMetadata.v_pixel_scale = texture->VPixelScale;
+        rawTexMetadata.type = texture->Type;
+        rawTexMetadata.resource = texture;
+
+        // #if _DEBUG && 0
+        // tex = reinterpret_cast<char*>(texture->imageData);
+        // ResourceMgr_GetNameByCRC(hash, fileName);
+        // fprintf(stderr, "G_SETTIMG_OTR_HASH: %s (%dx%d) size=0x%x, %08X\n", fileName, texture->Width,
+        //         texture->Height, texture->ImageDataSize, hash);
+        // #else
+        char* tex = NULL;
+        // #endif
+
+        // OTRTODO: We have disabled caching for now to fix a texture corruption issue with HD texture
+        // support. In doing so, there is a potential performance hit since we are not caching lookups. We
+        // need to do proper profiling to see whether or not it is worth it to keep the caching system.
+
+        tex = reinterpret_cast<char*>(texture->ImageData);
+
+        if (tex != nullptr) {
+            (*cmd0)--;
+            uintptr_t oldData = (*cmd0)->words.w1;
+            // TODO: wtf??
+            (*cmd0)->words.w1 = (uintptr_t)tex;
+
+            // if (ourHash != (uint64_t)-1) {
+            //     auto res = ResourceLoad(ourHash);
+            // }
+
+            (*cmd0)++;
+        }
+
+        (*cmd0)--;
+        Gfx* cmd = (*cmd0);
+        uint32_t fmt = C0(21, 3);
+        uint32_t size = C0(19, 2);
+        uint32_t width = C0(0, 10);
+
+        if (tex != NULL) {
+            gfx_dp_set_texture_image(fmt, size, width, fileName, texFlags, rawTexMetadata, tex);
         }
-        case G_FILLRECT:
+    } else {
+        SPDLOG_ERROR("G_SETTIMG_OTR_HASH: Texture is null");
+    }
+
+    (*cmd0)++;
+    return false;
+}
+
+// TODO handle special OTR opcodes later...
```

Comment:
this comment seems to be all over the place, is that something that will need to happen in every function it's sitting above?